src/math/p5.Vector.js

```diff
@@ -118,23 +127,63 @@ function vector(p5, fn) {
     // This is how it comes in with createVector()
     // This check if the first argument is a function
     constructor(...args) {
-      let x, y, z;
+      let dimensions = args.length; // TODO: make default 3 if no arguments
+      let values = args.map(arg => arg || 0);
       if (typeof args[0] === 'function') {
         this.isPInst = true;
         this._fromRadians = args[0];
         this._toRadians = args[1];
-        x = args[2] || 0;
-        y = args[3] || 0;
-        z = args[4] || 0;
-        // This is what we'll get with new p5.Vector()
+        values = args.slice(2).map(arg => arg || 0);
+      }
+      if (dimensions === 0) {
+        this.dimensions = 2;
+        this._values = [0, 0, 0];
+
+      } else {
+        this.dimensions = dimensions;
+        this._values = values;
+      }
+    }
+
+    get values() {
+      return this._values
+    }
+
+    set values(newValues) {
+      this._values = newValues.slice()
+    }
+
+    get x() {
```

Comment:
this allows us to still consume the vector x, y and z without having an explicit class attributes with that name, this making it more flexible
@limzykenneth